app/basicComponents/CopyButton.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import styled from 'styled-components';
+
+type Props = {
+  secondary?: boolean;
+  onClickCopy: React.MouseEventHandler<HTMLImageElement>;
+  children?: JSX.Element;
+  isCopied?: boolean;
+};
+
+const Wrapper = styled.div`
+  display: flex;
+`;
+
+const CopyIcon = styled.img.attrs<{ secondary?: boolean }>(
+  ({ theme: { icons }, secondary }) => ({
+    src: secondary ? icons.copySecondary : icons.copy,
+  })
+)<{ secondary?: boolean }>`
+  align-self: center;
+  width: 16px;
+  height: 15px;
+  margin: 0 3px;
+  cursor: pointer;
+  &:hover {
+    opacity: 0.5;
+  }
+  &:active {
+    transform: translate3d(2px, 2px, 0);
+  }
+`;
+
+const CopyButton = ({ onClickCopy, secondary, children, isCopied }: Props) => {
```

Comment:
Btw, instead of having all these states `copiedAddress` and `isCopied` in this hook, this `CopyButton` component can have also `onCopy: (value: string) => void` property, which will be called beside copying `value` to the clipboard.

Then, `AccountOverview` can have:
```
let t = 0;
const [lastCopiedAddress, setLastCopiedAddress] = useState<string | null>(null);
const handleCopyAddress = (address: string) => {
  await navigator.clipboard.writeText(address);
  setLastCopiedAddress(address):
  clearTimeout(t);
  t = setTimeout(() => setLastCopiedAddress(null), 5000);
};

// and then use <CopyButton value={...} onClick={handleCopyAddress} />
```
Which can be used to show this "COPIED" text ;)

p.s.
And then it seems like we don't need any special hook for this...

Reply:
Moved. We have different places where we showing text Copied text, and in case of reuse handler and not copying past it from place to place, not possible to avoid the hook or hide this functionality only in the button. 
What is possible is, to create 2 different components for different cases and use them with the internal handler for each component, and these handlers will be possible to replace with 1 hook and separate it from components folder and reuse for each component, for example `CopyButton` and `CopyTextSection`